### DISCUSSION TITLE
Built ONNX Runtime... now what?

### DISCUSSION THREAD

**Discussion by WilliamVenner** [microsoft/onnxruntime#22902] [Other Q&A] [answered]
I have built ONNX Runtime from source.

How do I "generate" the distributable form of it?

I.e., when I download ONNX Runtime from the releases page, I get a zip containing lib/ and include/

Directly inside include/ are some header files. Directly inside lib/ are some .so files.

However, include/ in the git repository of ONNX Runtime does not match the other include/ directory at all (the directory structure is much more complex than just a folder containing header files), and so the dependency that requires ONNX Runtime fails to include the header file.

Additionally, there are .so files under build/Linux/Release/ in my build, but I don't know which are relevant to the redistributable zip.

How do I generate the distributable form of ONNX Runtime? I don't see this mentioned in any documentation and it's difficult to figure out from the build scripts/Dockerfiles.

**Comment by skottmckay** [comment]
Why do you need to do this vs. using one of the release packages?

https://github.com/microsoft/onnxruntime/releases/ or platform specific package repositories like nuget, maven, cocoapods etc.

**Reply by WilliamVenner** [reply]
There aren't Linux aarch64 CUDA prebuilds, so I have to build from source.

**Reply by csukuangfj** [reply]
> There aren't Linux aarch64 CUDA prebuilds, so I have to build from source.
 
There are!

Please have a look at
- CUDA 11.4: https://github.com/csukuangfj/onnxruntime-libs/releases/download/v1.16.0/onnxruntime-linux-aarch64-gpu-1.16.0.tar.bz2
- CUDA 10.2: https://github.com/csukuangfj/onnxruntime-libs/releases/download/v1.11.0/onnxruntime-linux-aarch64-gpu-1.11.0.tar.bz2

We have tested it on Jetson Nano B01 and Jetson Orin NX with [sherpa-onnx](https://github.com/k2-fsa/sherpa-onnx)

**Comment by skottmckay** [comment, marked as answer]
All the packaging is done via CIs. You can probably refer to this: https://github.com/microsoft/onnxruntime/blob/main/tools/ci_build/github/azure-pipelines/templates/c-api-artifacts-package-and-publish-steps-windows.yml
